autoload/tagbar.vim

```diff
@@ -87,6 +87,124 @@ let s:warnings = {
     \ 'encoding': 0
 \ }
 
+let s:singular_types = {
+            \ 'Classes': 'Class',
```

Comment:
Thanks for the contribution @gryf. I just started to look into this and I like the concept of the finished feature but I think the implementation could use a little cleanup before its ready to merge.

For example this array seems like a huge amount of wasted code that doesn't accomplish much. It only changes a couple of types, most of them are pass-through. Would you by chance still be around to review and clean this up some?

(See issue 549 if you are wondering about the late action here! Hopefully we can work through some backlog.)

Reply:
It's just a map from plural form to singular one. This is used for indicating type of the object on the status bar - it would looks awkward if for example there would appear
```
~/somefile.py        MyClass (classes)
```
or
```
~/somefile.py        MyClass.__init__ (methods)
```
instead of `class` and `method`. I'm still around and I'm still actively using this little feature.

Reply:
Thanks, and I guess that makes sense. I still think the map is an eyesore, but the real crime here is committed by the English language with its mix of `+s`, `+es`, and even `y→ies` plural transformations. I'll let it slide. :wink: